config/daemonconfig/daemonconfig.go

```diff
@@ -136,6 +159,142 @@ func DumpConfigString(c interface{}) (string, error) {
 	return string(b), err
 }
 
+func IsDirExisted(path string) bool {
+	s, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return s.IsDir()
+}
+
+func makeStringSlice(slice []interface{}, cb func(string) string) ([]string, error) {
+	out := make([]string, len(slice))
+	for i, value := range slice {
+		str, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast %v to string", value)
+		}
+
+		if cb != nil {
+			out[i] = cb(str)
+		} else {
+			out[i] = str
+		}
+	}
+	return out, nil
+}
+
+func parseMirrorsConfigFromToml(b []byte) ([]MirrorConfig, error) {
+	var parsedMirrors []MirrorConfig
+
+	type HostFileConfig = hostFileConfig
+	c := struct {
+		HostConfigs map[string]hostFileConfig `toml:"host"`
+	}{}
+	tree, err := toml.LoadBytes(b)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse TOML: %w", err)
+	}
+	if err := tree.Unmarshal(&c); err != nil {
+		return nil, err
+	}
+
+	for key, value := range c.HostConfigs {
+		mirror := MirrorConfig{
+			Host:                key,
+			AuthThrough:         value.AuthThrough,
+			HealthCheckInterval: value.HealthCheckInterval,
+			FailureLimit:        value.FailureLimit,
+			PingURL:             value.PingURL,
+		}
+		if value.Header != nil {
+			header := http.Header{}
+			mirrorHeader := make(map[string]string, 0)
+			for key, ty := range value.Header {
+				switch v := ty.(type) {
+				case string:
+					header[key] = []string{v}
+				case []interface{}:
+					header[key], err = makeStringSlice(v, nil)
+					if err != nil {
+						return nil, err
+					}
+				default:
+					return nil, fmt.Errorf("invalid type %v for header %q", ty, key)
+				}
+				mirrorHeader[key] = header.Get(key)
```

Comment:
It will be more reliable if we can print a warning message here if there are more than one header values.

Reply:
Done.